lib/runtime-c-api/src/import/mod.rs

```diff
@@ -661,6 +682,53 @@ pub unsafe extern "C" fn wasmer_import_func_new(
     Box::into_raw(export) as *mut wasmer_import_func_t
 }
 
+/// Stop the execution of a host function, aka imported function. The
+/// function must be used _only_ inside a host function.
+///
+/// The pointer to `wasmer_instance_context_t` is received by the host
+/// function as its first argument. Just passing it to `ctx` is fine.
+///
+/// The error message must have a greater lifetime than the host
+/// function itself since the error is read outside the host function
+/// with `wasmer_last_error_message`.
+///
+/// This function returns `wasmer_result_t::WASMER_ERROR` if `ctx` or
+/// `error_message` are null.
+///
+/// This function never returns otherwise.
+#[no_mangle]
+#[allow(clippy::cast_ptr_alignment)]
+pub unsafe extern "C" fn wasmer_import_trap(
+    ctx: *const wasmer_instance_context_t,
+    error_message: *const c_char,
+) -> wasmer_result_t {
+    if ctx.is_null() {
+        update_last_error(CApiError {
+            msg: "ctx ptr is null in wasmer_import_trap".to_string(),
+        });
+
+        return wasmer_result_t::WASMER_ERROR;
+    }
+
+    if error_message.is_null() {
+        update_last_error(CApiError {
+            msg: "error_message is null in wasmer_import_trap".to_string(),
+        });
+
+        return wasmer_result_t::WASMER_ERROR;
+    }
+
+    let ctx = &*(ctx as *const Ctx);
+    let error_message = CStr::from_ptr(error_message).to_str().unwrap();
+
+    (&*ctx.module)
+        .runnable_module
+        .do_early_trap(Box::new(error_message)); // never returns
+
+    #[allow(unreachable_code)]
+    wasmer_result_t::WASMER_OK
```

Comment:
Perhaps we should panic here?  Seems like a small thing, not sure what best practice is here

Reply:
cbindgen doesn't generate a C bindgen for a Rust function with `!` as returned type.

Since I need to error before calling `do_early_trap`, I've decided to return a `wasmer_result_t`. And in the best scenario, I return `WASMER_OK`, even if it's unreachable. I'll add more documentation to explain.